libs/langchain-google-cloud-sql-pg/src/vectorStore.ts

```diff
@@ -293,6 +290,68 @@ class PostgresVectorStore extends VectorStore {
     const query = `DELETE FROM "${this.schemaName}"."${this.tableName}" WHERE ${this.idColumn} in (${idList})`;
     await this.engine.pool.raw(query);
   }
+
+  /**
+   * Searches for documents similar to a text query by embedding the query and
+   * performing a similarity search on the resulting vector.
+   *
+   * @param query - Text query for finding similar documents.
+   * @param k - Number of similar results to return. Defaults to 4.
+   * @param filter - Optional filter based on `FilterType`.
+   * @param _callbacks - Optional callbacks for monitoring search progress
+   * @returns A promise resolving to an array of `DocumentInterface` instances representing similar documents.
+   */
+  async similaritySearch(query: string, k: number = 4, filter?: this["FilterType"] | undefined, _callbacks?: Callbacks | undefined): Promise<Document[]> {
+    const embedding = await this.embeddings.embedQuery(query);
+    return await this.similaritySearchByVector(embedding, k, filter)
+  }
+
+  async similaritySearchByVector(embedding: number[], k?: number, filter?: this["FilterType"]): Promise<Document[]> {
+    const documents: Document[] = [];
+    const docsAndScores = await this.similaritySearchVectorWithScore(embedding, k, filter);
+    for (const [doc, _] of docsAndScores) {
+      documents.push(doc)
+    }
+    return documents;
+  }
+
+  async similaritySearchVectorWithScore(embedding: number[], k?: number, filter?: this["FilterType"]): Promise<[Document, number][]> {
+    const results = await this.queryCollection(embedding, k, filter)
+    let documentsWithScores:[Document, number][] = [];
+
+    for (const row of results) {
+      const metadata = (this.metadataJsonColumn && row[this.metadataJsonColumn]) ? row[this.metadataJsonColumn] : {};
+
+      for (const col of this.metadataColumns) {
+        metadata[col] = row[col];
+      }
+
+      documentsWithScores.push([
+        new Document({pageContent: row[this.contentColumn], metadata: metadata}),
+        row['distance']
+      ]);
+    }
+
+    return documentsWithScores;
+  }
+
+  private async queryCollection(embedding: number[], k?: Number | undefined, filter?: this["FilterType"] | undefined) {
+    k = k ?? this.k;
+    const operator = this.distanceStrategy.operator;
+    const searchFunction = this.distanceStrategy.searchFunction;
+    const _filter = filter !== undefined ? `WHERE "${Object.entries(filter)[0][0]}" = '${Object.entries(filter)[0][1]}'` : "";
+    let results;
+
+    const query = `SELECT *, ${searchFunction}("${this.embeddingColumn}", '[${embedding}]') as distance FROM "${this.schemaName}"."${this.tableName}" ${_filter} ORDER BY "${this.embeddingColumn}" ${operator} '[${embedding}]' LIMIT ${k};` 
```

Comment:
Instead of "*" (selecting all), we should only select the columns we need ie the id, content, metadata columns, and the metadata json column. They all need to have quotes around the column names too.